### ISSUE TITLE
Support for pandas DataFrame with DatetimeIndex

### ISSUE THREAD

**Issue by soxofaan** [cphyc/matplotlib-label-lines#6]
Hi, thanks for this great matplotlib "plugin"

I couldn't get it to work yet on pandas based plots with a DatetimeIndex:
<img width="647" alt="screen shot 2018-10-24 at 11 43 01" src="https://user-images.githubusercontent.com/44946/47422212-6c730e00-d782-11e8-9b28-8b59c4a7735c.png">

Any ideas, and is there any interest in supporting this?
I already had a quick look, but couldn't find a quick fix yet as date handling is quite messy territory

**Comment by cphyc** [comment]
Hey, thank you very much for your feedback.

I am not using labelled lines and dates personally so it would be hard for me to figure out a workaround. 

This should however be fixable by modifying the labelLine function (https://github.com/cphyc/matplotlib-label-lines/blob/81daa2d2e0af1b620bd76cb8257beb1749b34549/labellines/core.py#L37) and add support for datetimes.

This could be as simple as replacing https://github.com/cphyc/matplotlib-label-lines/blob/81daa2d2e0af1b620bd76cb8257beb1749b34549/labellines/core.py#L24 by 
```python
xdata = line.get_xdata()
if isinstance(xdata[0], datetime):
    xdate = [date2num(_) for _ in xdata]
```
and the same for ydata.

**Comment by soxofaan** [comment]
I found some time to work on this: see PR #8

**Comment by cphyc** [comment]
This should be fixed by #8 so I'm closing for now.
